server/src/main/java/org/opensearch/index/engine/CompositeIndexWriter.java

```diff
@@ -813,24 +794,15 @@ public final synchronized Iterable<Map.Entry<String, String>> getLiveCommitData(
 
     public void rollback() throws IOException {
         if (shouldClose()) {
-            Collection<IndexWriter> currentWriterSet = liveIndexWriterDeletesMap.current.criteriaBasedIndexWriterMap.values()
-                .stream()
-                .map(DisposableIndexWriter::getIndexWriter)
-                .collect(Collectors.toSet());
-
-            for (IndexWriter indexWriter : currentWriterSet) {
-                if (indexWriter.isOpen() == true) {
-                    indexWriter.rollback();
+            for (DisposableIndexWriter disposableIndexWriter : liveIndexWriterDeletesMap.current.criteriaBasedIndexWriterMap.values()) {
+                if (disposableIndexWriter.getIndexWriter().isOpen() == true) {
+                    disposableIndexWriter.getIndexWriter().rollback();
                 }
             }
 
-            Collection<IndexWriter> oldWriterSet = liveIndexWriterDeletesMap.old.criteriaBasedIndexWriterMap.values()
-                .stream()
-                .map(DisposableIndexWriter::getIndexWriter)
-                .collect(Collectors.toSet());
-            for (IndexWriter indexWriter : oldWriterSet) {
-                if (indexWriter.isOpen() == true) {
-                    indexWriter.rollback();
+            for (DisposableIndexWriter disposableIndexWriter : liveIndexWriterDeletesMap.old.criteriaBasedIndexWriterMap.values()) {
+                if (disposableIndexWriter.getIndexWriter().isOpen() == true) {
+                    disposableIndexWriter.getIndexWriter().rollback();
```

Comment:
This might not be thread-safe for instance the index writer might be closed while we are doing a rollback?

Reply:
Just wondering if requires to be thread safe. I mean if we are anyways closing writers here. In case the writers are already closed, we can move to closing next writes in the rollback flow. We can keep the entire code block in a try catch block and simply ignore AlreadyClosedException for the rollback flow?? 

In all other cases where any of the writer is closed, we can close all the writer as well?? As this means we encountered some tragic errors while writing to lucene